### PR TITLE
or_where conflict fix

### PR DESCRIPTION
**Pull request by cdunn**
Prevent or_where conflicts with pre-existing where filters on query by grouping into a new dynamic query

Ran into an issue using [user.last_name, user.first_name, user.id] as a cursor with a query that included a where filter for user.account_id == 123.  When using a before cursor, the or_where that is generated groups the pre-existing account_id = 123 filter into the initial where clause generated and thereby returning users on different accounts via the OR.

I think the test case reproduces it, let me know what you think

### CODE REVIEW

File: test/paginator_test.exs [hunk 1]
```diff
@@ -590,12 +616,12 @@ defmodule PaginatorTest do
     p1 = insert(:payment, customer: c2, charged_at: days_ago(11))
     p2 = insert(:payment, customer: c2, charged_at: days_ago(6))
     p3 = insert(:payment, customer: c2, charged_at: days_ago(8))
-    p4 = insert(:payment, customer: c2, charged_at: days_ago(12))
+    p4 = insert(:payment, customer: c2, amount: 2, charged_at: days_ago(12))
```
Comment (stevedomin) [inline test/paginator_test.exs]: Is this change still needed?

File: test/paginator_test.exs [hunk 2]
```diff
@@ -273,6 +273,32 @@ defmodule PaginatorTest do
...
+      customers: {c1, _c2, _c3},
+      payments: {_p1, _p2, _p3, _p4, _p5, p6, _p7, _p8, _p9, _p10, _p11, _p12}
+    } do
+      %Page{entries: [], metadata: _metadata} =
+        customer_payments_by_amount(c1)
+        |> Repo.paginate(
+          cursor_fields: [:amount, :charged_at, :id],
+          before: encode_cursor([p6.amount, p6.charged_at, p6.id]),
+          limit: 1
+        )
```
Comment (stevedomin) [inline test/paginator_test.exs]: Looks like we are missing an assertion here.

File: test/paginator_test.exs [hunk 3]
```diff
@@ -273,6 +273,32 @@ defmodule PaginatorTest do
     end
   end
 
+  describe "paginate a collection of payments with customer filter, sorting by customer name" do
```
Comment (stevedomin) [inline test/paginator_test.exs]: The last part of this description should read: "sorting by amount, charged_at"

File: test/paginator_test.exs (hunk 1 above)
Comment (cdunn) [reply test/paginator_test.exs]: Kinda...but only in that it will ensure the test on line 288 fails if any pre-existing query filtering is ever broken

Comment (stevedomin) [reply test/paginator_test.exs]: Makes sense.

**Comment by stevedomin** [comment]
Thank you so much @cdunn !!

**Comment by stevedomin** [comment]
Release 0.3.1 is out with the fix: https://hex.pm/packages/paginator/0.3.1
